### PR TITLE
Adding support for a hiding limit to practice climb

### PR DESCRIPTION
**Pull request by jayRyan24**
Adding support for a max hiding count. This takes care of #1245.

### CODE REVIEW

File: crossing-training.lic [hunk 1]
```diff
@@ -671,9 +671,11 @@ class CrossingTraining
       check_osrel unless @researching
       check_research
       retreat
-      loop do
+      attempts_to_hide = 5
+      while attempts_to_hide > 0 do
```
Comment (rpherbig) [inline crossing-training.lic]: Do you need to decrement `attempts_to_hide` in the loop?

Comment (jayRyan24) [inline crossing-training.lic]: YES OF COURSE I DO... :(

Comment (jayRyan24) [inline crossing-training.lic]: Sorry, my hate. Thanks for the catch.

File: crossing-training.lic [hunk 2]
```diff
@@ -671,9 +671,12 @@ class CrossingTraining
...
         break if hide?
+        attempts_to_hide - 1
```
Comment (rpherbig) [inline crossing-training.lic]: This doesn't assign back to `attempts_to_hide`, so I think this will still be an infinite loop... can you double check?

Comment (jayRyan24) [inline crossing-training.lic]: Ah, was told it did, lol. I will check!

**Pull request merged**